Tasks/FuncToolsInstallerV0/src/functoolsinstaller.ts

```diff
@@ -0,0 +1,34 @@
+"use strict";
+
+import * as tl from 'azure-pipelines-task-lib/task';
+import * as path from 'path';
+import * as toolLib from 'azure-pipelines-tool-lib/tool';
+import * as utils from './utils';
+
+tl.setResourcePath(path.join(__dirname, '..', 'task.json'));
+
+async function downloadFuncTools() {
+    const version = await utils.getFuncToolsVersion();
+    const funcToolsPath = await utils.downloadFuncTools(version);
+
+    // prepend the tools path. instructs the agent to prepend for future tasks
+    if (!process.env['PATH'].startsWith(path.dirname(funcToolsPath))) {
```

Comment:
Instead of `startsWith` shouldn't it be `contains`

Reply:
No..in the task code we do `which`, and we want it to find `this version` of the tool. If a tool is present in `multiple paths` `which` resolves to the first entry. So `contains` may not work.

Reply:
If there are multiple tools being installed in the same pipeline, the `PATH` var probably wouldn't have the func tool in the beginning.

Reply:
Agreed. But this is the best we can do to ensure that the most recent installation (in the pipeline) is available for the tasks that follow. Hypothetical example:

`Install func 1.0.0`
`Deploy function 1`
`Install func 2.0.0`
`Deploy function 2`
`Install func 1.0.0`
`Deploy function 3`

In such a case, with `contains` check, `Deploy function 3` would actually get `func tool` version `2.0.0` (since its prepended in the `PATH` after `1.0.0`).

Agree that in above example, if third step was `Install kubectl 1.0.0`, prepending PATH in the 5th step would be redundant. Although one point to note is that the user is contributing to this redundancy.